pkg/streamingpromql/functions.go

```diff
@@ -61,6 +61,37 @@ func SingleInputVectorFunctionOperatorFactory(name string, f functions.FunctionO
 	}
 }
 
+func TimeFunctionOperatorFactory(name string, f functions.FunctionOverInstantVectorDefinition) InstantVectorFunctionOperatorFactory {
```

Comment:
Given this is only used by `TimeTransformationFunctionOperatorFactory`, I would move this inside that function - I don't think the separation is useful, and they're always used together.

Reply:
Should be inlined in 16843e7ab802f2662b2f5bde70dabc8bb4cdb2ce